CONTRIBUTING.md

```diff
@@ -0,0 +1,39 @@
+# Contributing to Bearded.Utilities
+
+You want to help us improve our little library? ~~That's awesome!~~ You're awesome!
+
+Whether you have a feature request, found a bug, want to add some code of your own or even just have a question - all of these are very welcome.
+
+Our library is pretty small, so let's try and go easy on the rules. :)
+
+
+## The basics
+
+- **Have a question?** [Open an issue!](https://github.com/beardgame/utilities/issues/new/choose)
+- **Found a bug?** [Open an issue!](https://github.com/beardgame/utilities/issues/new/choose)
+- **Would you like to request a feature, change or improvement?** [Open an issue!](https://github.com/beardgame/utilities/issues/new/choose)
+
+You could also always check out list of issues and see if there's any discussions you want to share your thoughts on.
+
+
+## You'd actually like to make changes yourself?
+
+Wow, that's even better!
+
+In principle this as easy as forking the repository, making some changes and [opening a pull request](https://github.com/beardgame/utilities/compare). From there, we'll review your PR as soon as we can manage, ask questions or request changes if necessary, and once we're happy merge it into our main branch. It's that easy!
+
+If you want to know ahead of time if we'd welcome a certain feature or change, you could always create an issue where you propose that change and mention that you'd like to implement it yourself. We'll let you know what we think as soon as we can. The main reason we might decline code is if we feel it doesn't fit with the purpose of our libraries, but we're very open to suggestions and good arguments.
```

Comment:
While the probability of duplicate work with the amount of traffic currently is minimal, I am wondering whether we should ask people to always assign themselves to an issue when they start work on something. 

OpenTK works with draft PRs, which I think isn’t a bad idea either, since it is as simple as pushing a branch. However, it works badly with the PR template since the author will have to go back and update the initial post, which as a reviewer is also not gonna be great, especially since it adds a bunch of conversational noise that should maybe be on a FR issue instead.

All in all, maybe having nothing is fine, but in most cases we should recommend an assigned issue for all in flight work.

(Not a blocker for PR approval)

Reply:
Will mention issue assignments, I think that's good enough. I'm not a big fan of draft PRs overall, or at least not of mandatory ones since it requires pushing before I've finished my work, which in the case of small issues is an additional effort for little benefit. For bigger changes, sure, makes sense to use them to get feedback, but it's not like we have so much traffic coming through here that it matters for filtering purposes.